terraform/modules/github-ci-bootstrap/main.tf

```
@@ -34,40 +44,70 @@ locals {
   # Use provided bucket name or computed default
   terraform_state_bucket = coalesce(var.terraform_state_bucket, local.default_bucket_name)
 
-  # Flatten target_projects into individual service permissions
+  # Flatten target_projects into individual service permissions for read-write access
   project_service_permissions = flatten([
     for project_id, config in var.target_projects : [
       for service in config.required_services : {
         key        = "${project_id}-${service}"
         project_id = project_id
         service    = service
-        role       = local.service_roles[service]
+        role       = local.read_write_roles[service]
       }
     ]
   ])
+
+  # Flatten target_projects into individual service permissions for read-only access
+  project_service_permissions_readonly = flatten([
```

```suggestion
  project_service_permissions_ro = flatten([
```
(And it's good to be consistent in the terminology you use too.)